types/urllib/index.d.ts

```diff
@@ -0,0 +1,253 @@
+ *
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ */
+export function request(url: string | url.URL, options?: RequestOptions): Promise<any>;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param callback @see Callback
+ */
+export function request(url: string | url.URL, callback: Callback): void;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ * @param callback @see Callback
+ */
+export function request(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+
+/**
+ * Handle request with a callback.
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param callback @see Callback
+ */
+export function requestWithCallback(url: string | url.URL, callback: Callback): void;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ * @param callback @see Callback
+ */
+export function requestWithCallback(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+
+/**
+ * yield urllib.requestThunk(url, args)
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ */
+export function requestThunk(url: string | url.URL, options: RequestOptions): (callback: (...args: any[]) => void) => void;
+
+/**
+ * alias to request.
+ * Handle all http request, both http and https support well.
+ *
+ * @example
+ * // GET http://httptest.cnodejs.net
+ * urllib.request('http://httptest.cnodejs.net/test/get', function(err, data, res) {});
+ * // POST http://httptest.cnodejs.net
+ * var args = { type: 'post', data: { foo: 'bar' } };
+ * urllib.request('http://httptest.cnodejs.net/test/post', args, function(err, data, res) {});
+ *
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ */
+export function curl(url: string | url.URL, options?: RequestOptions): Promise<any>;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param callback @see Callback
+ */
+export function curl(url: string | url.URL, callback: Callback): void;
+/**
+ * @param url The URL to request, either a String or a Object that return by url.parse.
+ * @param options Optional, @see RequestOptions.
+ * @param callback @see Callback
+ */
+export function curl(url: string | url.URL, options: RequestOptions, callback: Callback): void;
+/**
+ * The default request timeout(in milliseconds).
+ * @const
+ */
+export const TIMEOUT: number;
+/**
+ * The default request & response timeout(in milliseconds).
+ * @const
```

Comment:
I think the `@const` tag is also redundant given that this is a `const` declaration -- will have to update the `no-redundant-jsdoc` lint rule to test for that.

Reply:
OK, I'll remove redundant tags and fix the comment, and there's nothing important in `README.md`, just my habit. Leave it alone if there's no need to remove it.